resolver_test.go

```diff
@@ -12,6 +12,29 @@ import (
 	"github.com/nbio/st"
 )
 
+func CheckTXT(t *testing.T, domain string) {
+	r := NewResolver()
+	rrs, err := r.ResolveErr(domain, "TXT")
+	st.Expect(t, err, nil)
+
+	rrs2, err := net.LookupTXT(domain)
+	st.Expect(t, err, nil)
+	for _, rr := range rrs2 {
+		exsists := false
+		for _, rr2 := range rrs {
+			if rr2.Type == "TXT" && rr == rr2.Value {
+				exsists = true
+			}
+		}
+		if !exsists {
+			t.Errorf("TXT record %q not found", rr)
+		}
+	}
+	if count(rrs, func(rr RR) bool { return rr.Type == "TXT" }) != len(rrs2) {
+		t.Errorf("TXT record count mismatch: %d != %d", count(rrs, func(rr RR) bool { return rr.Type == "TXT" }), len(rrs2))
+	}
```

Comment:
nit: dedupe count execution

```suggestion
	c := count(rrs, func(rr RR) bool { return rr.Type == "TXT" })
	if c != len(rrs2) {
		t.Errorf("TXT record count mismatch: %d != %d", c, len(rrs2))
	}
```